library/alloc/src/vec/mod.rs

```diff
@@ -1023,8 +1023,11 @@ impl<T, A: Allocator> Vec<T, A> {
 
     /// Shrinks the capacity of the vector as much as possible.
     ///
-    /// It will drop down as close as possible to the length but the allocator
-    /// may still inform the vector that there is space for a few more elements.
+    /// The behavior of this method depends on the allocator being used. If the allocator supports
+    /// in-place shrinking, some or all of the excess capacity may be returned to the allocator.
+    /// If not, this method may cause the vector's data to be copied to a new allocation, although
+    /// no guarantees are given that this new allocation is actually smaller. See
+    /// [`Allocator::shrink`] for more details.
```

Comment:
This is correct but occasionally some users misinterpret "no guarantees" and start doing silly things trying to get better behavior.

Instead we can say that the behavior depends on the allocator which may either shrink in-place or reallocate and that the resulting allocation may still have some excess capacity just as `with_capacity` is allowed to.

So it reads more like "performs like copy into Vec::with_capacity(), or better".

Reply:
Thanks, the rephrase does make it sound a little less dire 😄